doc(.github): add pr template

OliverHoenig opened the pull request:
This adds a template for pull requests. 
It is meant to be to standardize the process of pull requests and to
 help the author and the reviewer during this process.

OliverHoenig committed:
doc(.github): add pr template

This adds a template for pull requests. 
It is meant to be to standardize the process of pull requests and to
 help the author and the reviewer during this process.

OliverHoenig committed:
doc(pull_request_template.md): link to commit conventions

So that the reviewer can easily refer and check if the conventions of the commit messages are met, a link to the conventions is added.

OliverHoenig commented:
Hello @csvtuda, 
this is the implementation of the checklists for PRs that you have already reviewed. Could you please check the implementation and accept the PR? Many thanks in advance.

csvtuda commented on .github/.github/pull_request_template.md:
Should the file's path really be `.github/.github/pull_request_template.md`?

OliverHoenig replied:
No, you're right, it should just be `.github/pull_request_template.md`. Take a look at the open discussions and when those are closed, I'll adjust that. (I'm not sure what will happen to the pending suggestions if I move the file now)

OliverHoenig replied:
I moved the file into the correct directory now.

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+## Before submitting the PR, please make sure you do the following
+
+### Prepare Code for PR
+* [ ] The automated tests were executed locally (`mvn clean test`), and it was ensured that they passed.
```
```suggestion
* [ ] The automated tests were successfully executed locally (`mvn clean test`).
```

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+* [ ] The automated tests were executed locally (`mvn clean test`), and it was ensured that they passed.
+* [ ] All commented-out code and debug statements have been removed.
+* [ ] A test is included that fails without this PR but passes with it.
+* [ ] The relevant documentation, including code comments and public documentation, has been updated.
```
```suggestion
* [ ] The relevant documentation, including code comments and public documentation, has been updated in a corresponding PR.
```

I think it does not make sense to update the productive documentation before the changes actually go live.

OliverHoenig replied:
Yes, I can partly understand that. But in my opinion, it definitely makes sense to update the documentation in the code. I think the public documentation should also be adapted as soon as possible. However, is it difficult to guarantee that the check will be carried out later. What do you think?
You can't guarantee at this point that the documentation will be adapted at a late stage. The idea here was to only allow changes to the production code once the documentation has also been adapted. But it true, now you might create a public documentation that is ahead of the actual code, which wouldn't be good either.

OliverHoenig replied:
```suggestion
* [ ] The documentation in the code has been updated and corresponds to the team guidelines.
* [ ] A person has been assigned to update the public documentation as soon as possible if this PR requires changes to it.
```

What do you think about this instead of the suggestion above?

csvtuda replied:
> But it true, now you might create a public documentation that is ahead of the actual code, which wouldn't be good either.

I usually create a new PR in the documentation repository and merge them when the new version is released as well.

> What do you think about this instead of the suggestion above?

That's better!

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+    * [ ] [Qytera Github-Page (github.io)](https://github.com/Qytera-Gmbh/Qytera-Gmbh.github.io)
+
+
+### PR Description
+* [ ] A message body is added that provides context and guidance so that the reviewer can quickly understand what the PR does. In the PR body, this includes. In the PR body, includes:
+    * [ ] The purpose of the pull request, clearly illustrating what problem this PR solves and why this solution is appropriate.
+    * [ ] An overview of the changes, ideally with 1-2 examples that illustrate the changes.
+    * [ ] Links to any additional context such as tracking issues or tickets.
```
```suggestion
```

I'd simply include that in the description above (see https://github.com/Qytera-Gmbh/QTAF/pull/325#discussion_r1679122687).

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+# 📄 PR Description
+
+[Add a description for this pull request. Follow the checklist below.] 
```
```suggestion
<!--
Add a description for this pull request. Follow the checklist below.
- What is the purpose of the pull request?
- Illustrate the problem this PR solves and why this solution is appropriate.
  - Include an overview of the changes, ideally with 1-2 examples that illustrate the changes.
  - Include links to any additional context such as tracking issues or tickets.
-->
```

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+<!--- Describe your changes in detail including a relevant motivation and context for this change -->
+
+
+# ✒️ Checkliste PR-Author
```
```suggestion
# ✒️ Checklist PR author
```

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+* [ ] Review from appropriate team members is requested.
+- Be prepared to explain and discuss changes.
+
+# 🔍 Checkliste PR-Reviewer
```
```suggestion
# 🔍 Checklist  PR reviewer
```

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+### Understand the Context
+* [ ] The PR description is read thoroughly to understand the changes.
+* [ ] It is verified that the changes fulfill the purpose described in the PR.
+* [ ] If the PR belongs to an issue or is intended to represent a user story, it is verified that the requirements and acceptance criteria are met. (If necessary, the author is asked where these were specified.)
```
```suggestion
* [ ] If the PR belongs to an issue or is intended to represent a user story, it is verified that the requirements and acceptance criteria are met.
```

Redundant with the subsequent bullet point.

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+
+### Code Quality, Style & Functionality
+* [ ] It is ensured that the code adheres to the projects' [coding standards and style guides](https://qytera.atlassian.net/wiki/spaces/QO/pages/3800727592/QTAF+-+Coding+Standards).
+* [ ] The code is clean, readable and well-organized code. 
```
```suggestion
* [ ] The code is clean, readable and well-organized. 
```

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+* [ ] The code is clean, readable and well-organized code. 
+* [ ] The commits are named according to [the team conventions](https://qytera.atlassian.net/wiki/spaces/WIS/pages/3861348360/Commit+und+PR+Konventionen+f+r+QTAF). (If not, the author is informed so that he/she can improve for the next time. There are no changes to the commit history required.)
+* [ ] It is ensured that there are no commented-out code or debug statements left.
+* [ ] Potential edge cases and error handling are checked. (Are potential errors properly retained? Is there an error handling strategy?)
```
```suggestion
* [ ] The changed or added code correctly handles potential edge cases and errors.
```

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+
+### Review Preparation
+* [ ] Review from appropriate team members is requested.
+- Be prepared to explain and discuss changes.
```
```suggestion
```

I think this is unnecessary. It doesn't add any value (you wouldn't create a PR if you weren't ready for feedback).

csvtuda commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+### Code Quality, Style & Functionality
+* [ ] It is ensured that the code adheres to the projects' [coding standards and style guides](https://qytera.atlassian.net/wiki/spaces/QO/pages/3800727592/QTAF+-+Coding+Standards).
+* [ ] The code is clean, readable and well-organized code. 
+* [ ] The commits are named according to [the team conventions](https://qytera.atlassian.net/wiki/spaces/WIS/pages/3861348360/Commit+und+PR+Konventionen+f+r+QTAF). (If not, the author is informed so that he/she can improve for the next time. There are no changes to the commit history required.)
```
This is an internal link whose contentshould probably be moved to a [`CONTRIBUTING.md`](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) of some sort. But since nobody but Qytera is developing QTAF anyways, I think this is fine for now.

OliverHoenig replied:
Yes, that is correct. I also think that it will work for now.

OliverHoenig committed:
docs(PR-teamplate): shorten sentence

Co-authored-by: Basti <18561435+csvtuda@users.noreply.github.com>

OliverHoenig commented on .github/.github/pull_request_template.md:
```diff
@@ -0,0 +1,54 @@
+* [ ] The automated tests were successfully executed locally (`mvn clean test`).
+* [ ] All commented-out code and debug statements have been removed.
+* [ ] A test is included that fails without this PR but passes with it.
+* [ ] The relevant documentation, including code comments and public documentation, has been updated.
```
```suggestion
* [ ] The documentation in the code has been updated and corresponds to the team guidelines
* [ ] A person has been assigned to update the public documentation as soon as possible if this PR requires changes to it
```

What do you think about this instead of the suggestion above?

OliverHoenig committed:
docs(PR-Template): remove checks for description

Co-authored-by: Basti <18561435+csvtuda@users.noreply.github.com>

OliverHoenig committed:
docs(PR-Template): add a more detailed desciption

Co-authored-by: Basti <18561435+csvtuda@users.noreply.github.com>

OliverHoenig committed:
docs(PR-Template): remove redundant comment

Co-authored-by: Basti <18561435+csvtuda@users.noreply.github.com>

OliverHoenig committed:
docs(PR-Template): simplify wording

Co-authored-by: Basti <18561435+csvtuda@users.noreply.github.com>

OliverHoenig committed:
docs(PR-Template): remove unnecessary comment

Co-authored-by: Basti <18561435+csvtuda@users.noreply.github.com>

OliverHoenig committed:
docs(PR-Temaple): fix typo

Co-authored-by: Basti <18561435+csvtuda@users.noreply.github.com>

OliverHoenig committed:
docs(PR-Temaple): fix typo

Co-authored-by: Basti <18561435+csvtuda@users.noreply.github.com>

OliverHoenig committed:
docs(PR-Temaple): fix typo

Co-authored-by: Basti <18561435+csvtuda@users.noreply.github.com>

OliverHoenig committed:
docs(PR-Template): improve check for updating docu

There was a conflict with the check for updating the official documentation. If the documentation gets updated before the code, then the documentation is ahead of the actual code. 
If the documentation gets updated after the code, then it is not possible to confirm at this point in time that the documentation has already been updated.

OliverHoenig committed:
docs(PR-Template): correct dir

OliverHoenig commented:
@csvtuda If there aren't any more changes required, please confirm the PR. Thanks for your help.

OliverHoenig committed:
docs(PR-Temp): Delete .github/.github directory
